rocAL/source/readers/image/file_source_reader.cpp

```diff
@@ -141,42 +142,41 @@ void FileSourceReader::reset() {
 }
 
 Reader::Status FileSourceReader::subfolder_reading() {
-    if ((_sub_dir = opendir(_folder_path.c_str())) == nullptr)
-        THROW("FileReader ShardID [" + TOSTR(_shard_id) + "] ERROR: Failed opening the directory at " + _folder_path);
-
     std::vector<std::string> entry_name_list;
-    std::string _full_path = _folder_path;
-
-    while ((_entity = readdir(_sub_dir)) != nullptr) {
-        std::string entry_name(_entity->d_name);
-        if (strcmp(_entity->d_name, ".") == 0 || strcmp(_entity->d_name, "..") == 0) continue;
-        entry_name_list.push_back(entry_name);
-    }
-    closedir(_sub_dir);
-    std::sort(entry_name_list.begin(), entry_name_list.end());
-
     auto ret = Reader::Status::OK;
-    for (unsigned dir_count = 0; dir_count < entry_name_list.size(); ++dir_count) {
-        std::string subfolder_path = _full_path + "/" + entry_name_list[dir_count];
-        filesys::path pathObj(subfolder_path);
-        if (filesys::exists(pathObj) && filesys::is_regular_file(pathObj)) {
-            // ignore files with non-image extensions
-            auto file_extension_idx = subfolder_path.find_last_of(".");
-            if (file_extension_idx != std::string::npos) {
-                std::string file_extension = subfolder_path.substr(file_extension_idx + 1);
-                std::transform(file_extension.begin(), file_extension.end(), file_extension.begin(),
```

Comment:
Do we not need to check the extensions?
Please check this before removing

Reply:
I am afraid this will break our existing pipeline. Please schedule a call with Sundar and myself.

Reply:
I tested it with file source reader using CMD in our c++ pipeline - it works as expected -  ./rocal_unittests 13 /media/MIVisionX-data/rocal_data/images_jpg/labels_folder/ abc 1000 1000 0 0 1 0 1